docs/design/features/readytorun-elf-support-roadmap.md

```diff
@@ -0,0 +1,166 @@
+Roadmap to native Linux executable format (ELF) support in Ready to Run
+=======================================================================
+
+Revisions:
+* 1.1 - [Tomas Rylek](https://github.com/trylek) - 2/27/2020
+
+# Introduction
+
+This document details the steps needed for supporting native Linux executable
+format (ELF) for ready-to-run binaries on Linux. Today, ready-to-run binaries
+use the Windows PE executable format on all three supported platforms
+(Windows, Linux, OSX). Due to this fact ready-to-run binaries are "unnatural"
+on non-Windows platforms incurring both performance and user experience costs
+(incomplete integration with OS-native application management, inability to
+use OS executable loader, lack of OS-native diagnostic experience). This
```

Comment:
There are different gradiations of 'OS-native' diagnostics. I am guessing we are proposing only a very scoped set of improvements at this point? An example progression given increasing levels of effort:
1. Use ELF format - tools such as readelf and basic module enumeration in debuggers work
2. Add ELF dynamic symbol table - debuggers/profilers should be able to resolve IPs in those methods to names
3. Add DWARF information for all methods + line information - debuggers/profilers can resolve all pre-compiled methods to names and source, jitted code is still opaque
4. .NET type information encoded with DWARF symbols - Limited object inspection works. Functionality that isn't C-like still probably doesn't work such as variable length types (string/array) or generic instantiations not known at image compile time. 
5. .Net specific DWARF extensions + 1st class support for .Net built into standard tools - Everything works : )

Reply:
It's a shame that the GCC LTO information isn't stable and depends heavily on GIMPLE (GCC's IL); otherwise Crossgen2 could emit it and we could get LTO "for free". (Don't know about Clang's approach here, but I would imagine it depends heavily on the LLVM IR.)

Reply:
> GCC LTO information

A few problems with it:
- The intermediate representation of C/C++ compilers is not aware of GC and it is missing other constructs required to support managed code well. Same problems exist with clang.
- GCC has incompatible license